VMBackup/main/HttpUtil.py

```diff
@@ -33,23 +33,28 @@
 
 class HttpUtil(object):
     """description of class"""
-    def __init__(self, hutil):
-        Config = None
-        self.proxyHost = None
-        self.proxyPort = None
-        try:
-            waagent.MyDistro = waagent.GetMyDistro()
-            Config = waagent.ConfigurationProvider(None)
-        except Exception as e:
-            errorMsg = "Failed to construct ConfigurationProvider, which may due to the old wala code with error: %s, stack trace: %s" % (str(e), traceback.format_exc())
-            hutil.log(errorMsg)
-            Config = None
-        self.logger = hutil
-        if Config != None:
-            self.proxyHost = Config.get("HttpProxy.Host")
-            self.proxyPort = Config.get("HttpProxy.Port")
-
-        self.tmpFile = './tmp_file_FD76C85E-406F-4CFA-8EB0-CF18B123365C'
+    __instance = None
+    """Singleton class initialization"""
+    def __new__(cls, hutil):
+        if(cls.__instance is None):
+            hutil.log("Creating HttpUtil")
+            cls.__instance = super(HttpUtil, cls).__new__(cls)
+            try:
+                waagent.MyDistro = waagent.GetMyDistro()
+                Config = waagent.ConfigurationProvider(None)
+            except Exception as e:
+                errorMsg = "Failed to construct ConfigurationProvider, which may due to the old wala code with error: %s, stack trace: %s" % (str(e), traceback.format_exc())
+                hutil.log(errorMsg)
+                Config = None
+            cls.__instance.logger = hutil
+            if Config != None:
+                cls.__instance.proxyHost = Config.get("HttpProxy.Host")
+                cls.__instance.proxyPort = Config.get("HttpProxy.Port")
+            cls.__instance.tmpFile = './tmp_file_FD76C85E-406F-4CFA-8EB0-CF18B123365C'
+        else:
+            cls.__instance.logger = hutil
```

Comment:
can be removed

Reply:
Kept it intentionally. On customer VM, without this, logger was trying to write to file (enforce flag change was not reflecting).

Reply:
actually logger will still try to write. if we are using hutil.log.
enforce flag is above hutil.log
I am not sure how it is working.